Adding updating threat modelling text (#657)
cncf/tag-security: security-whitepaper/cloud-native-security-whitepaper.md

## Threat Modeling

> For organizations adopting cloud native, a primary mechanism for identifying risks, controls and mitigations is to perform threat modeling. While there are many threat modeling techniques they share several core characteristics. All start with building a scoped representation of a system's architecture. This begins with identifying all important processes, data stores, and [security boundaries](https://www.oreilly.com/library/view/cissp-certified-information/9780470276884/9780470276884_security_boundaries.html#:~:text=A%20security%20boundary%20is%20the,a%20LAN%20and%20the%20Internet.). Once boundaries have been established and the relevant elements of the system partitioned within those boundaries, the next step is to model how these elements interact with special attention paid to any interactions that cross security boundaries. 

Reviewer: ```suggestion
For organizations adopting cloud native, a primary mechanism for identifying risks, controls and mitigations is to perform threat modeling. While there are many threat modeling techniques they share several core characteristics. All start with building a scoped representation of a system's architecture. This begins with identifying all important processes, data stores, and [security boundaries](https://www.oreilly.com/library/view/cissp-certified-information/9780470276884/9780470276884_security_boundaries.html). Once boundaries have been established and the relevant elements of the system are partitioned within them, the next step is to model how these elements interact with special attention paid to any interactions that cross security boundaries. 
```

Remove anchor text highlight of link, minor nits

> The below guidance is an enhancement of the four step OWASP threat modeling recommended for cloud native capabilities.

Reviewer: ```suggestion
The below guidance is an enhancement of the four step [OWASP threat modeling](https://owasp.org/www-community/Threat_Modeling) recommended for cloud native capabilities.
```

### End-to-end architecture

A clear understanding of the organization's or individual's cloud native architecture should result in data impact guidance and classifications. This helps teams organize data distribution within the architecture as well as the additional protection mechanisms for it later on. cloud native diagrams and documentation should not only include the core components of the overall system design but should also take into consideration the location of the source code, the buckets and other storage mechanisms in use, and any additional aspects of the software development cycle. These are all areas that must be considered when initiating threat modeling for cloud native.

### Threat Identification

> When considering threats specific to an organization's cloud native capabilities, it is recommended to leverage a mature, well-used model of threats such as STRIDE or OCTAVE. Common threats organizations may wish to consider for their cloud native architectures includes, but is not limited to:

Reviewer: is there a reason for removing links here?

Author: Looks like at some point someone stripped them when creating the cnsmap doc, and when I copied the new content from there, that propagated here. I'll fix both.

- Denial of Service (DoS) resulting from a pod that does not have resource limits applied therefore consumes the entire node level CPU and memory, worker node is then lost
- Elevation of privilege could happen if a pod is running with unrestricted or higher privileged pod security policy or by modifying the security context of a pod or a container

Threat actors to consider for cloud native security are consistent with existing threat modeling practices:

- Malicious insider - An actor with malicious intent and with authorization to perform actions within the modeled system.

> - Uninformed insider - An actor with authorization to perform actions within the modeled system (assuming anyone can be duped).

Reviewer 2: Is `spoofed` instead of `duped` better representation of the intent here?

Author: Great question. My initial reaction is that "duped" may be better as it leaves this open to any sort of accidental or confused action that a user might be compelled to take. To my mind, this is different to Spoofing which is usually more about assuming a specific role (implicit or explicit). That said; within the (clunky) STRIDE framework, it does fit into spoofing more than any other category. 

I think I'm comfortable with "duped" as "User was duped by spearphishing and clicked a CSRF link" is clearly the type of issue we're concerned with for an "Uniformed insider".

All that said, super happy to change it if you feel strongly about it.

Reviewer 3: I think it is ok as is.  If you decide to change it, might consider adding more words to explain the two cases explained in this comment, something like "assuming anyone can be deceived or make an error with unintended results"

> - Uninformed insider - An actor that with authorization to perform actions within the modeled system (assume anyone can be duped).

Reviewer: ```suggestion
- Uninformed insider - An actor with authorization to perform actions within the modeled system (assuming anyone can be duped).
```

> - Malicious outsider - An actor outside of the system. They could be attacking via the internet, via supply chain, via physical perimeter etc.

Reviewer: ```suggestion
- Malicious outsider - An actor outside of the system with malicious intent, able to launch attacks via the internet, supply chain, physical perimeter etc.
```

> - Malicious outsider - An actor outside of the system with malicious intent, able to launch attacks via the internet, supply chain, physical perimeter etc.

Reviewer 2: Maybe worth rephrasing to clarify that the outsider does not have access to the modeled system:

`An actor outside of the system with malicious intent, able to launch attacks via the internet, supply chain, physical perimeter etc. without explicit authorization to perform actions within the modeled system`

Author: I'll make that change.

> There are other actors that may interact with the modeled system (e.g Uninformed outsiders) and that can be included for completeness. It's likely that controls for their actions will be a subset of those for the primary actors listed above.

Reviewer: ```suggestion
There are other actors that may interact with the modeled system (e.g uninformed outsiders) and they can be included for completeness. It's likely that controls for their actions will be a subset of those for the primary actors listed above.
```
